docs/index.html

```diff
@@ -1354,24 +1354,26 @@ <h3 id="name-semantics-6">
         </h3>
 <p id="section-2.5-1">The root selector <code>$</code> not only selects the root node of the input document, but it also produces as output a list consisting of one node: the input document.<a href="#section-2.5-1" class="pilcrow">¶</a></p>
 <p id="section-2.5-2">A matcher may match a node and, if it matches, may then select zero or more descendants of the node for further processing.
-               But there's more to it than this: each matcher acts on a list of nodes and produces a list of nodes, as follows.<a href="#section-2.5-2" class="pilcrow">¶</a></p>
-<p id="section-2.5-3">After the root selector, the remainder of the JSON Path is processed by passing lists of nodes from one matcher to the next ending up with a list of nodes which is the result of
-               matching the JSON Path to the input JSON document.<a href="#section-2.5-3" class="pilcrow">¶</a></p>
-<p id="section-2.5-4">Each matcher acts on its input list of nodes as follows. For each node in the list, the matcher may or may not match the node. If the matcher
+               A syntactically valid matcher MUST NOT produce errors. This means that some operations which might be considered erroneous, such as indexing beyond the end of an array,
+               simply result in fewer descendants being selected.<a href="#section-2.5-2" class="pilcrow">¶</a></p>
+<p id="section-2.5-3">But a matcher doesn't just act on a single node: each matcher acts on a list of nodes and produces a list of nodes, as follows.<a href="#section-2.5-3" class="pilcrow">¶</a></p>
+<p id="section-2.5-4">After the root selector, the remainder of the JSON Path is processed by passing lists of nodes from one matcher to the next ending up with a list of nodes which is the result of
+               matching the JSON Path to the input JSON document.<a href="#section-2.5-4" class="pilcrow">¶</a></p>
+<p id="section-2.5-5">Each matcher acts on its input list of nodes as follows. For each node in the list, the matcher may or may not match the node. If the matcher
                matches the node, it selects zero or more descendants of the node.
```

Comment:
> zero or more descendants of the node

Are we sure we're not going to allow nodes from input list to get included in the output list (like `descendant-or-self::` axis in XPath)?

Reply:
Ah, I think you missed the following in the Terminology section 2.2:
> The descendants of a node are taken to include the node itself and the descendants of any children of the node.

Reply:
Yep. But in this case it seems that the term "descendant" doesn't fit very well. What would you think about replacing it with "branch"?

Reply:
I don't follow why "descendant" doesn't fit well. Please could you elaborate?

("branch" seems like a strange term, unless you have a particular definition in mind.)

Reply:
Yes, I take your point. It's too risky to redefine the normal meaning of descendant as people could miss, or forget, the redefinition. I'll try rewording this.

Reply:
Done. See what you think.